pkg/controller/migplan/registry_health.go

```diff
@@ -84,60 +84,63 @@ func (r *registryHealth) run() {
 				return
 			}
 
-			srcClient, err := srcCluster.GetClient(r.hostClient)
-			if err != nil {
-				log.Trace(err)
-				return
-			}
-			destClient, err := destCluster.GetClient(r.hostClient)
-			if err != nil {
-				log.Trace(err)
-				return
-			}
+			if srcCluster.Status.IsReady() {
+				srcClient, err := srcCluster.GetClient(r.hostClient)
+				if err != nil {
+					log.Trace(err)
+					return
+				}
 
-			registryPodsSrc, err := r.getRegistryPods(plan, srcClient)
-			if err != nil {
-				log.Trace(err)
-				return
-			}
+				registryPodsSrc, err := r.getRegistryPods(plan, srcClient)
+				if err != nil {
+					log.Trace(err)
+					return
+				}
 
-			registryPodsDest, err := r.getRegistryPods(plan, destClient)
-			if err != nil {
-				log.Trace(err)
-				return
+				for _, registryPod := range registryPodsSrc.Items{
+					if !registryPod.Status.ContainerStatuses[0].Ready {
+						r.enqueue(plan)
+						continue
+					}
+				}
 			}
 
-			for _, registryPod := range registryPodsSrc.Items{
-				if health.ContainerUnhealthy(registryPod) || health.UnhealthyStatusConditions(registryPod) {
-					r.enqueue(plan)
-					continue
+			if destCluster.Status.IsReady() {
+				destClient, err := destCluster.GetClient(r.hostClient)
+				if err != nil {
+					log.Trace(err)
+					return
+				}
+
+				registryPodsDest, err := r.getRegistryPods(plan, destClient)
+				if err != nil {
+					log.Trace(err)
+					return
 				}
-			}
 
-			for _, registryPod := range registryPodsDest.Items{
-				if health.ContainerUnhealthy(registryPod) || health.UnhealthyStatusConditions(registryPod) {
-					r.enqueue(plan)
-					continue
+				for _, registryPod := range registryPodsDest.Items{
+					if !registryPod.Status.ContainerStatuses[0].Ready {
+						r.enqueue(plan)
+						continue
+					}
 				}
 			}
+
+
 		}
 
 	}
+//TODO: need see if this go routine should be stopped and returned
 
 }
 
 func (r *registryHealth) getRegistryPods(plan v1alpha1.MigPlan, registryClient compat.Client) (corev1.PodList, error){
 
-	registryDeployment, err := plan.GetRegistryDeployment(registryClient)
-	if err != nil {
-		log.Trace(err)
-		return corev1.PodList{}, err
-	}
 	registryPodList := corev1.PodList{}
-	err = registryClient.List(context.TODO(), k8sclient.MatchingLabels(registryDeployment.GetLabels()), &registryPodList)
+	err := registryClient.List(context.TODO(), k8sclient.MatchingLabels(plan.GetLabels()), &registryPodList)
```

Comment:
I wonder if this will work, we need to list the pod based on "name": plan.Name label selectore. The plan.GetLabels() does not return this

might have to do something like `v1.SetAsLabelSelector(map[string]string{"name": plan.Name})`

Reply:
Updated the List call